File: app/src/main/java/ru/MeatGames/roguelike/tomb/StatsDB.java
package ru.MeatGames.roguelike.tomb;

/**
 * Класс базы данных характеристик игрока.
 */

public class StatsDB {
	/**
	 * @uml.property  name="a"
	 */
	public int a; // значение
	public String n;
	/**
	 * @uml.property  name="s"
	 */
	public boolean s; // true если значение статично (защита, скорость и т.д.)
	/**
	 * @uml.property  name="m"
	 */
	public boolean m; // true если значение максимально/минимально (здоровье, мана и т.д.), тогда n <= n + 1
	// s и m == false значит, что эта характеристика является максимумом для характеристики с предыдущем номером
}

File: app/src/main/java/ru/MeatGames/roguelike/tomb/StatsDB.kt
package ru.MeatGames.roguelike.tomb

class StatsDB {

    var value: Int = 0
    var title: String? = null
    var single: Boolean = false // true означает, что эта характеристика не имеет ограничение на максимальное значение
    var maximum: Boolean = false /* true означает, что эта характеристика является текущим значением следующей характеристики
    (например, текущее здоровье), т.е. имеет ограничение на максимальное значение*/
    // single и maximum == false означает, что эта характеристика является максимальным значением для предыдущей характеристики

}